src/components/plots/areaPlot.ts

```diff
@@ -18,14 +18,19 @@ export module Plot {
      */
     constructor(xScale: Scale.AbstractQuantitative<X>, yScale: Scale.AbstractQuantitative<number>) {
       super(xScale, yScale);
-      this.classed("area-plot", true);
-      this.project("y0", 0, yScale); // default
+      _Util.Methods.uniqAdd(this._cssClasses, "area-plot");
 
-      this.animator("reset", new Animator.Null());
-      this.animator("main", new Animator.Base()
-                                        .duration(600)
-                                        .easing("exp-in-out"));
+      this._animators["reset"] = new Animator.Null();
+      this._animators["main"] = new Animator.Base()
+                                            .duration(600)
+                                            .easing("exp-in-out");
       this._defaultFillColor = new Scale.Color().range()[0];
+
+      this._projections["y0"] = {
+        accessor: d3.functor(0),
+        scale: yScale,
+        attribute: "y0"
+      };
```

Comment:
I believe I am ok with removing scales from the constructor, but since that cleanup has not been made yet, this is what I'm doing for now until that cleanup is made.

The reason why I'm doing this instead of project is because this is significantly more explicit on what is happening to the state of the object:  Items in the constructor should only use the inputs given to them from the constructor; it should not require any inputs from other instance variables, and calling .project does not guarantee this.